Gemfile

```diff
@@ -4,7 +4,7 @@ gem 'rubocop', '>= 1.0', '< 2.0'
 
 git_source(:github) { |repo| "https://github.com/#{repo}.git" }
 
-ruby '3.0.3'
+ruby '3.1.1'
```

Comment:
- [ ] Kindly add these gems to your Gemfile because all your tests failed while testing your controllers. After I added these gems, some of your tests passed while some failed due to wrong assertions. The first gem is for window users and the second gem is for controller testing when you're testing for templates (render_template). There are very important gems to have so that your tests can run on any OS and test templates successfully.

```ruby
    gem 'ffi'
    gem 'rails-controller-testing'
```